### PR TITLE
[material-ui] Adds linting

### PR DESCRIPTION
**Pull request by nbgraham**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>

I was attempting [another pull request](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18179) to material-ui and realized that there was no lint file. So I created it and then tried to fix some of the lint errors.

I had to ignore some types of errors so that the build would pass because I wasn't sure of the best way to fix them:
[tslint.json](https://github.com/nbgraham/DefinitelyTyped/blob/material_ui_lint/types/material-ui/tslint.json):
```
{ 
    "extends": "dtslint/dt.json",
    "rules": {
        "strict-export-declare-modifiers": false,
        "no-empty-interface": false,
        "dt-header": false,
        "ban-types": false
    }
}
```

### CODE REVIEW

File: types/material-ui/index.d.ts [hunk 1]
```diff
@@ -5862,8 +5860,8 @@ declare module 'material-ui/svg-icons/av/library-books' {
 }
 
 declare module 'material-ui/svg-icons/av/art-track' {
-    export import AvArtTrack = __MaterialUI.SvgIcon;
-    export default AvArtTrack;
+    export import AlettTrack = __MaterialUI.SvgIcon;
+    export default AlettTrack;
```
Comment (alitaheri) [inline types/material-ui/index.d.ts]: var -> let replace mismatch 😅

File: types/material-ui/index.d.ts [hunk 2]
```diff
@@ -7881,7 +7879,7 @@ declare module "material-ui/svg-icons" {
     export import AvAddToQueue = __MaterialUI.SvgIcon; // require('material-ui/svg-icons/av/add-to-queue');
     export import AvAirplay = __MaterialUI.SvgIcon; // require('material-ui/svg-icons/av/airplay');
     export import AvAlbum = __MaterialUI.SvgIcon; // require('material-ui/svg-icons/av/album');
-    export import AvArtTrack = __MaterialUI.SvgIcon; // require('material-ui/svg-icons/av/art-track');
+    export import AlettTrack = __MaterialUI.SvgIcon; // require('material-ui/svg-icons/av/art-track');
```
Comment (alitaheri) [inline types/material-ui/index.d.ts]: Same here

**Review by alitaheri — CHANGES_REQUESTED** [review]

File: types/material-ui/index.d.ts (hunk 1 above)
Comment (nbgraham) [inline types/material-ui/index.d.ts]: Haha nice catch! I fixed it with https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18189/commits/c46fa7bb9aee52301e14fe2d7ca4861f2a5d8efa

File: types/material-ui/index.d.ts (hunk 2 above)
Comment (nbgraham) [inline types/material-ui/index.d.ts]: Thanks! https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18189/commits/c46fa7bb9aee52301e14fe2d7ca4861f2a5d8efa

**Review by alitaheri — APPROVED** [review]
Nice job ❤️  👍 🎉

**Pull request merged**